rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,666 @@
+
+
+### Operator signed in
+At the start of their shift, operators need to confirm that they are fit-for-duty and do not have any electronic devices. They currently do this by physically signing a paper trainsheet: in the future, they will do this digitally.
+
+Event type: `com.mbta.ctd.glides.operator_signed_in.v1`
+Fields in the event:
+- `author` (Author): the inspector who signed in the operator
+- `operator` (string): the badge number of the operator who signed in
+- `signedInAt` (RFC3999 timestamp): the time at which they signed in (separate from the `time` of the event)
+- `confirmation` (string): how the operator confirmed that they signed in. Current possibilities are `"type:<badge number>"` if they signed by typing in their badge number, or `"tap"` if they signed in by tapping their badge on an RFID reader, but future string formats may be added without warning, and consumers SHOULD NOT depend on any specific format.
+
+The badge's RFID serial number is not included in the stream for security reasons.
+
+### Trips Updated
+Glides has new information about trips. Consumers who want to know what trains are running should pay attention to this event, and don't need to pay attention to any other events.
+
+Event type: `com.mbta.ctd.glides.trips_updated.v1`
+Fields in the event:
+- `author` (Author): the inspector who inputed the new information.
+- `inputType` (string): the action the inspector performed in Glides that caused the update. There are many ways to update trips in Glides, and all updated information normalized into the same format in `tripUpdates`. This field indicates what the inspector was doing in Glides, for consumers who care not just about service, but also care about how inspectors are using Glides. Example values are `"add-trip"` or `"manage-headways"`, but no guarantees are given about what strings will be used.
+- `tripUpdates` (array of ([TripUpdated](#TripUpdated)|[TripAdded](#TripAdded))): The list of one or more trips that have new information.
+
+#### TripUpdated
+This indicates that a trip was updated in some fashion:  consist, operators, departure time. The trip is either a scheduled trip, or an added trip that has already appeared in the events stream. Fields which are not present are not considered to be updated.
+
+Fields in the object:
+- `type` (string): `"updated"|"added"`. Determines whether this is a `TripUpdated` object or a `TripAdded` object. If it's `"added"`, then it's a `TripAdded` object, see below. Subsequent updates to previously-added trips have `type` `"updated"`.
+- `tripKey` (Trip Key): which trip is being updated.
+- `comment` (string, optional): free text information about the trip.
+- `startLocation` (Location, optional): the new destination of the train.
+- `endLocation` (Location, optional): the new destination of the train.
+- `startTime` (Time, optional): if present, the new time that the train is expected to depart `startLocation` (or the existing `startLocation` of the trip).
```

Comment:
After chatting with Melody, we went with `"none"` (actively removed) and `"unset"` (pretend we never updated it, go back to schedule), because they mean different things, and this is clearer than having two meanings of `null`. There are no places where those string constants could be mixed up with string data.